### PR TITLE
[MLIR] Add the convergent attribute to LLVM Dialect

### PR DESCRIPTION
**Pull request by FMarno**
In order to use the convergent attribute in the GPUToLLVMSPV pass, I've added the attribute to the LLVM dialect.
Some details on the convergent attribute https://llvm.org/docs/ConvergentOperations.html#convergent-operations

### CODE REVIEW

File: mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td [hunk 1]
```diff
@@ -1425,6 +1425,7 @@ def LLVM_LLVMFuncOp : LLVM_Op<"func", [
     UnitAttr:$dso_local,
     DefaultValuedAttr<CConv, "CConv::C">:$CConv,
     OptionalAttr<SymbolRefAttr>:$comdat,
+    OptionalAttr<UnitAttr>:$convergent,
```
Comment (victor-eds) [inline mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td:1428]:
```suggestion
    UnitAttr:$convergent,
```
I think this is the standard way. Not sure why other `UnitAttr` are `Optional` in this operation definition.

**Review by victor-eds — COMMENTED** [review]
Can we have a test in `mlir/test/Dialect/LLVMIR/func.mlir`? Also, modifying the operation assembly format in `mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp` (`FuncOp::parse/print`) would be better than the current syntax IMO.

**Review by gysit — APPROVED** [review]
LGTM

**Comment by FMarno** [comment]
> Also, modifying the operation assembly format in `mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp` (`FuncOp::parse/print`) would be better than the current syntax IMO.

There current syntax is in line with other attributes like `no_inline`, `always_inline`, or `alignment`, which seems appropriate to me.

**Comment by victor-eds** [comment]
> > Also, modifying the operation assembly format in `mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp` (`FuncOp::parse/print`) would be better than the current syntax IMO.
> 
> There current syntax is in line with other attributes like `no_inline`, `always_inline`, or `alignment`, which seems appropriate to me. examples in this test:
> 
> https://github.com/llvm/llvm-project/blob/a4ea8a48367203b5d4c2fbab49f88ad7f31ab03d/mlir/test/Target/LLVMIR/Import/function-attributes.ll#L349-L357

Seems fair. Thanks! We'd only be missing an export to LLVM IR test then

**Comment by FMarno** [comment]
> Seems fair. Thanks! We'd only be missing an export to LLVM IR test then

Thank you for the suggestion, that's been added now

**Review by victor-eds — APPROVED** [review]

**Pull request merged**